src/main/java/echoserver/EchoServer.java

```diff
@@ -0,0 +1,60 @@
+package echoserver;
+
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Scanner;
+
+public class EchoServer implements Runnable {
+    private Scanner input;
+    private PrintWriter output;
+    private Socket clientSocket;
+
+    public EchoServer(Socket clientSocket) {
+        this.clientSocket = clientSocket;
+    }
+
+    public void run() {
+        try {
+            setServerInputOutput(clientSocket);
+            output.println("Connected to server");
+            echo();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void echo() {
+        String inputString;
+        while ((inputString = SocketIO.readFromInputStream(input)) != null) {
+            if (InputValidator.isQuit(inputString)) {
```

Comment:
There appear to be a few static/class methods that get used in `EchoServer`, and I think you'll find that this makes Java code especially difficult to test (Mockito doesn't have a good way to mock this out from a test, short of doing something drastic with the JVM).

Why not pass in an _instance_ of `InputValidator` instead?  You may find that the flexibility of being able to provide different instances (initially for the sake of testing) also adds a lot of flexibility to this code (then it could support multiple languages, for example).